### PR TITLE
bigquery/language/logging/prediction/vision: decouple packages

### PR DESCRIPTION
**Pull request by stephenplusplus**
After #1529 ships, this does the de-coupling of packages using the new method, `common.util.isCustomType()`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "docs": "node ./scripts/docs/packages.js",
     "bundle": "node ./scripts/docs/bundle.js",
     "lint": "jshint scripts/ packages/ system-test/ test/ && jscs packages/ system-test/ test/",
-    "test": "npm run docs && npm run bundle && mocha test/docs.js packages/*/test/*.js",
+    "test": "npm run docs && npm run bundle && mocha --timeout 5000 test/docs.js packages/*/test/*.js",
```
Comment (callmehiphop) [inline package.json]: What happened to `--no-timeouts`?

Comment (stephenplusplus) [inline package.json]: We have that for system tests, but not unit.

File: packages/logging/package.json [hunk 2]
```diff
@@ -52,16 +52,16 @@
...
-    "@google-cloud/common": "^0.1.0",
-    "@google-cloud/pubsub": "^0.1.0",
-    "@google-cloud/storage": "^0.1.0",
     "google-proto-files": "^0.2.1",
     "is": "^3.0.1",
     "string-format-obj": "^1.0.0"
   },
   "devDependencies": {
     "async": "^1.4.2",
+    "@google-cloud/bigquery": "*",
```
Comment (callmehiphop) [inline packages/logging/package.json]: Why do we need these on dev dependencies?

Comment (callmehiphop) [inline packages/logging/package.json]: system tests?

Comment (stephenplusplus) [inline packages/logging/package.json]: Yep.

File: packages/vision/package.json [hunk 3]
```diff
@@ -53,8 +53,7 @@
     "arrify": "^1.0.0",
     "async": "^1.4.2",
     "extend": "^3.0.0",
-    "@google-cloud/common": "^0.1.0",
-    "@google-cloud/storage": "^0.1.0",
```
Comment (callmehiphop) [inline packages/vision/package.json]: I'm assuming there is a system test for this? Don't we still need it as a dev dependency?

File: packages/logging/package.json [hunk 4]
```diff
@@ -52,16 +52,16 @@
...
+    "@google-cloud/common": "*",
```
Comment (callmehiphop) [inline packages/logging/package.json]: Also I believe that @google-cloud/common needs to stay a regular `dependency`.

Comment (stephenplusplus) [inline packages/logging/package.json]: Woops.

File: packages/vision/package.json (hunk 3 above)
Comment (stephenplusplus) [inline packages/vision/package.json]: There wasn't a system test that uses a file actually. Should I write one?

Comment (callmehiphop) [inline packages/vision/package.json]: Assuming there is code that uses storage, I think that would be a great idea!

Comment (stephenplusplus) [inline packages/vision/package.json]: Done!

**Comment by stephenplusplus** [comment]
PTAL.

**Pull request merged**